nft_ingester/src/utils/logs.rs

```diff
@@ -0,0 +1,62 @@
+use flatbuffers::{ForwardsUOffset, Vector};
+use lazy_static::lazy_static;
+use regex::Regex;
+use solana_sdk::pubkey::Pubkey;
+
+pub fn filter_events_from_logs(log_messages: &Vec<&str>) -> Result<Vec<String>, ()> {
+    lazy_static! {
+        static ref CLRE: Regex = Regex::new(
+            r"Program data: ((?:[A-Za-z\d+/]{4})*(?:[A-Za-z\d+/]{3}=|[A-Za-z\d+/]{2}==)?$)"
+        )
+        .unwrap();
+    }
+    let mut events: Vec<String> = vec![];
+
+    for line in log_messages {
+        let line_str = String::from(*line);
+        let captures = CLRE.captures(&line_str);
+        let b64raw = captures.and_then(|c| c.get(1)).map(|c| c.as_str());
+        b64raw.map(|raw| events.push((raw).parse().unwrap()));
+    }
+    if events.is_empty() {
+        println!("No events captured!");
+        Err(())
+    } else {
+        Ok(events)
+    }
+}
+
+pub fn parse_logs<'a>(
+    log_messages: Option<Vector<'a, ForwardsUOffset<&str>>>,
+) -> Result<Vec<(Pubkey, Vec<&'a str>)>, ()> {
```

Comment:
Nice! looks like this was able to convert to references fairly straightforwardly using the lifetime specifiers to bind the input to the output Vec, and not have to change much of the function's logic.  Hopefully it was just add the `'a` and not have too many extraneous issues.